From 40ddbf453cfe2bb04a237d15a70c6cd261a4aed8 Mon Sep 17 00:00:00 2001
From: Eric Sun <jian.sun@freescale.com>
Date: Thu, 2 Feb 2012 18:59:20 +0800
Subject: [PATCH 0942/2463] ENGR00173659 MX6Q_UART Change Physical / Virtural
 Port mapping

For ARM2 and Sabreauto, change TTY0 to TTY3 (which is physical UART4)
For SabreSD, Change TTY3 to TTY0 (which is physical UART1)

Mapping Changed as the following
Physical	Virtual
--------	--------
1		0
2		1
3		2
4		3

Signed-off-by: Eric Sun <jian.sun@freescale.com>
---
 arch/arm/mach-mx6/board-mx6q_arm2.c           |    2 +-
 arch/arm/mach-mx6/board-mx6q_sabresd.c        |    2 +-
 arch/arm/plat-mxc/devices/platform-imx-uart.c |    6 +++---
 3 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/arch/arm/mach-mx6/board-mx6q_arm2.c b/arch/arm/mach-mx6/board-mx6q_arm2.c
index 497cc23..7c655af 100644
--- a/arch/arm/mach-mx6/board-mx6q_arm2.c
+++ b/arch/arm/mach-mx6/board-mx6q_arm2.c
@@ -519,7 +519,7 @@ static const struct imxuart_platform_data mx6q_uart1_data __initconst = {
 
 static inline void mx6q_arm2_init_uart(void)
 {
-	imx6q_add_imx_uart(0, NULL);
+	imx6q_add_imx_uart(3, NULL);
 	imx6q_add_imx_uart(1, &mx6q_uart1_data);
 }
 
diff --git a/arch/arm/mach-mx6/board-mx6q_sabresd.c b/arch/arm/mach-mx6/board-mx6q_sabresd.c
index fa81c74..724fd15 100644
--- a/arch/arm/mach-mx6/board-mx6q_sabresd.c
+++ b/arch/arm/mach-mx6/board-mx6q_sabresd.c
@@ -362,7 +362,7 @@ static const struct anatop_thermal_platform_data
 static inline void mx6q_sabresd_init_uart(void)
 {
 	imx6q_add_imx_uart(2, NULL);
-	imx6q_add_imx_uart(3, NULL);
+	imx6q_add_imx_uart(0, NULL);
 }
 
 static int mx6q_sabresd_fec_phy_init(struct phy_device *phydev)
diff --git a/arch/arm/plat-mxc/devices/platform-imx-uart.c b/arch/arm/plat-mxc/devices/platform-imx-uart.c
index f51f69e..ecfadc3 100644
--- a/arch/arm/plat-mxc/devices/platform-imx-uart.c
+++ b/arch/arm/plat-mxc/devices/platform-imx-uart.c
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2009-2010 Pengutronix
+ * Copyright (C) 2009-2012 Pengutronix
  * Uwe Kleine-Koenig <u.kleine-koenig@pengutronix.de>
  *
  * This program is free software; you can redistribute it and/or modify it under
@@ -130,10 +130,10 @@ const struct imx_imx_uart_1irq_data imx53_imx_uart_data[] __initconst = {
 const struct imx_imx_uart_1irq_data imx6q_imx_uart_data[] __initconst = {
 #define imx6q_imx_uart_data_entry(_id, _hwid)				\
 	imx_imx_uart_1irq_data_entry(MX6Q, _id, _hwid, SZ_4K)
-	imx6q_imx_uart_data_entry(0, 4),
+	imx6q_imx_uart_data_entry(0, 1),
 	imx6q_imx_uart_data_entry(1, 2),
 	imx6q_imx_uart_data_entry(2, 3),
-	imx6q_imx_uart_data_entry(3, 1),
+	imx6q_imx_uart_data_entry(3, 4),
 };
 #endif /* ifdef CONFIG_SOC_IMX6Q */
 
-- 
1.7.7.4

